### PR TITLE
Use valid Psbt for serde regression test

### PR DESCRIPTION
**Pull request by Subhra264**
The `unknown` key used in the `Psbt` (in `serde_regression_psbt` test) is not really unknown. Also the `preimages` in the `Input` don't seem to be valid. Though they don't affect the tests because of the `serde::serialize`, the `Psbt` itself is invalid and hence this PR.

### CODE REVIEW

File: bitcoin/tests/serde.rs [hunk 1]
```diff
@@ -312,6 +312,10 @@ fn serde_regression_psbt() {
         }],
     };
 
+    // Check if the psbt is valid
+    let serialized = psbt.serialize();
+    Psbt::deserialize(&serialized).unwrap();
```
Comment (tcharding) [inline bitcoin/tests/serde.rs:317]: This doesn't do validity checks, it just proves we can roundtrip serialize/deserialize the PSBT according to bip-174, right?

Perhaps
```
/// Sanity, check we can roundtrip BIP-174 serialize.
```

**Comment by tcharding** [comment]
Also, thanks for improving the tests!

File: bitcoin/tests/serde.rs (hunk 1 above)
Comment (Subhra264) [reply bitcoin/tests/serde.rs:317]: Yeah! makes much more sense! (I really need to improve my commenting skills :))

File: bitcoin/tests/serde.rs [hunk 2]
```diff
@@ -245,7 +245,7 @@ fn serde_regression_psbt() {
         }],
     };
     let unknown: BTreeMap<raw::Key, Vec<u8>> =
-        vec![(raw::Key { type_value: 1, key: vec![0, 1] }, vec![3, 4, 5])].into_iter().collect();
+        vec![(raw::Key { type_value: 9, key: vec![0, 1] }, vec![3, 4, 5])].into_iter().collect();
```
[the remark this answers is not in the document]
Comment (Subhra264) [reply bitcoin/tests/serde.rs:248]: Only in the case of Input, it means "Proof-of-reserves commitment" which is not supported for now in rust-bitcoin. For the rest, it does not yet exist and just avoids confusion with other valid key-value pairs while serializing/deserializing.

**Comment by Subhra264** [comment]
For some reason, the commit `[407dec0bcedd24245e60195c09c61b5832582b2e](https://github.com/rust-bitcoin/rust-bitcoin/commit/407dec0bcedd24245e60195c09c61b5832582b2e)` in master did not pass the CI checks even though [here](https://github.com/rust-bitcoin/rust-bitcoin/pull/2024) it says they passed.

**Comment by Subhra264** [comment]
Ohh! just saw https://github.com/rust-bitcoin/rust-bitcoin/pull/2029

**Comment by tcharding** [comment]
Your CI fail is fixed on master now, if you rebase I can ack, this is good to go in IMO. Sorry for taking so long, I overlooked this one.

**Review by tcharding — APPROVED** [review]
ACK 95b7a95fc28f4a84df6e10d514f3538752319420

**Review by apoelstra — APPROVED** [review]
ACK 95b7a95fc28f4a84df6e10d514f3538752319420

**Pull request merged**